core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -756,6 +757,15 @@ private DefaultFunctionResolver subtime() {
     );
   }
 
+  /**
+   * Extracts a date, time, or datetime from the given string.
+   * It accomplishes this using another string which specifies the input format.
+   */
+  private DefaultFunctionResolver str_to_date() {
+    return define(BuiltinFunctionName.STR_TO_DATE.getName(),
+        impl(nullMissingHandling(DateTimeFunction::exprStrToDate), DATETIME, STRING, STRING));
```

Comment:
FP is required to build `DATETIME` if user specifies only time input.

Reply:
I opted to use a default value instead of '0001-01-01' based on how I've addressed some other comments.  Please let me know if this is acceptable. Using function properties is possible, but I am not sure if it is very useful for this function if you are simply using it to extract `time` data from a string.

Reply:
Unfortunately, our return value is always full `DATETIME`.
For example, `str_to_date('10 20 30', '%H %i %s')` should return `10:20:30` of today's date. Please add tests for this too.

Reply:
Fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/231/commits/8d19f213ef49e4695a8f0889e1dbb0a846d0042a